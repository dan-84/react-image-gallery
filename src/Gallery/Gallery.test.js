import React from 'react'
import Gallery from './index'
import renderer from 'react-test-renderer'
import Image from 'gatsby-image'

const fluidShapeMock = path => ({
  aspectRatio: 1.5,
  src: path,
  srcSet: `some srcSet`,
  srcSetWebp: `some srcSetWebp`,
  sizes: `(max-width: 600px) 100vw, 600px`,
  base64: `string_of_base64`,
})

describe('Gallery component', () => {
  test('that it renders with empty props', () => {
    const component = renderer.create(<Gallery images={[]} thumbs={[]} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('that it renders with data in images', () => {
    const component = renderer.create(
      <Gallery
        images={[
          '/images/image001.jpg',
          '/images/image002.jpg',
          '/images/image003.jpg',
          '/images/image004.jpg',
        ]}
        thumbs={[]}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('that it renders with data in thumbs', () => {
    const component = renderer.create(
      <Gallery
        images={[]}
        thumbs={[
          { fluid: fluidShapeMock('/thumb/image001.jpg') },
          { fluid: fluidShapeMock('/thumb/image002.jpg') },
          { fluid: fluidShapeMock('/thumb/image003.jpg') },
          { fluid: fluidShapeMock('/thumb/image004.jpg') },
          { fluid: fluidShapeMock('/thumb/image005.jpg') },
        ]}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('that it renders with data in images & thumbs', () => {
    const component = renderer.create(
      <Gallery
        images={[
          '/images/image001.jpg',
          '/images/image002.jpg',
          '/images/image003.jpg',
          '/images/image004.jpg',
        ]}
        thumbs={[
          { fluid: fluidShapeMock('/thumb/image001.jpg') },
          { fluid: fluidShapeMock('/thumb/image002.jpg') },
          { fluid: fluidShapeMock('/thumb/image003.jpg') },
          { fluid: fluidShapeMock('/thumb/image004.jpg') },
          { fluid: fluidShapeMock('/thumb/image005.jpg') },
        ]}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
