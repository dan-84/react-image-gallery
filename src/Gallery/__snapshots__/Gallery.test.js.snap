// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component that it renders with a custom column class 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Gallery component that it renders with data in images & thumbs 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Gallery component that it renders with data in images 1`] = `
<div
  className="row"
/>
`;

exports[`Gallery component that it renders with data in thumbs 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-6 px-0"
    onClick={[Function]}
  >
    <div
      className="m-1"
    >
      <div
        className="rounded gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Gallery component that it renders with empty props 1`] = `
<div
  className="row"
/>
`;
